File: backend/src/main/java/com/team05/petmeeting/global/security/util/RefreshTokenUtil.java
package com.team05.petmeeting.global.security.util;

import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.stereotype.Component;

@Component
public class RefreshTokenUtil {

    public static final String REFRESH_TOKEN_COOKIE_NAME = "refreshToken";
    private static final int REFRESH_TOKEN_MAX_AGE = 7 * 24 * 60 * 60; // 7일

    /**
     * RefreshToken 쿠키 생성 ( 재발급 )
     **/
    public void add(HttpServletResponse response, String refreshToken) {
        Cookie cookie = new Cookie(REFRESH_TOKEN_COOKIE_NAME, refreshToken);

        cookie.setHttpOnly(true);
        cookie.setSecure(true); // HTTPS 환경에서만
        cookie.setAttribute("SameSite", "Strict");
        cookie.setPath("/"); // 필요하면 "/auth/refresh"로 좁혀도 됨
        cookie.setMaxAge(REFRESH_TOKEN_MAX_AGE);

        response.addCookie(cookie);
    }

    /**
     * RefreshToken 쿠키 삭제
     */
    public void delete(HttpServletResponse response) {
        Cookie cookie = new Cookie(REFRESH_TOKEN_COOKIE_NAME, null);

        cookie.setHttpOnly(true);
        cookie.setSecure(true);
        cookie.setPath("/");
        cookie.setMaxAge(0);

        response.addCookie(cookie);
    }
}

File: backend/src/main/java/com/team05/petmeeting/global/security/util/RefreshTokenUtil.kt
package com.team05.petmeeting.global.security.util

import jakarta.servlet.http.Cookie
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component

@Component
class RefreshTokenUtil {

    fun add(response: HttpServletResponse, refreshToken: String) {
        Cookie(REFRESH_TOKEN_COOKIE_NAME, refreshToken).apply {
            isHttpOnly = true
            secure = true
            setAttribute("SameSite", "Strict")
            path = "/"
            maxAge = REFRESH_TOKEN_MAX_AGE
        }.also(response::addCookie)
    }

    fun delete(response: HttpServletResponse) {
        Cookie(REFRESH_TOKEN_COOKIE_NAME, null).apply {
            isHttpOnly = true
            secure = true
            path = "/"
            maxAge = 0
        }.also(response::addCookie)
    }

    companion object {
        const val REFRESH_TOKEN_COOKIE_NAME = "refreshToken"
        private const val REFRESH_TOKEN_MAX_AGE = 7 * 24 * 60 * 60
    }
}
